formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonNamesMap.kt

```diff
@@ -99,11 +99,16 @@ internal fun SerialDescriptor.getJsonNameIndexOrThrow(json: Json, name: String,
 internal inline fun Json.tryCoerceValue(
     elementDescriptor: SerialDescriptor,
     peekNull: () -> Boolean,
+    tryPeekNull: () -> Boolean,
```

Comment:
I'd merge these two lambdas to lambda with boolean param because they have the same implementation for tree decoders and differ only in boolean parameter for string decoder. You can add parameter name for lambda to readability: `peekNull: (consume: Boolean) -> Boolean`